main.go

```diff
@@ -102,6 +109,43 @@ func main() {
 	}
 }
 
+func runFifo(ctx context.Context, cfg notifyConfig) error {
+	if flag.NArg() != 2 {
+		flag.Usage()
+		os.Exit(2)
+	}
+	fifoPath := flag.Arg(1)
+
+	// Open pipe with O_NONBLOCK to ensure we don't get stuck here and
+	// miss the first write
+	p, err := os.OpenFile(fifoPath, os.O_RDONLY|syscall.O_NONBLOCK, 0601)
```

Comment:
Probably should be mode `0600` (or even `0400`, since we're only reading) unless I'm missing something obvious here:

```suggestion
	p, err := os.OpenFile(fifoPath, os.O_RDONLY|syscall.O_NONBLOCK, 0600)
```

Reply:
Good point.. I think I copy pasted this from somewhere. `0400` Should probably work